server/data/courseClient.ts

```diff
@@ -15,6 +15,17 @@ export default class CourseClient {
     return (await this.restClient.get({ path: apiPaths.courses.index({}) })) as Array<Course>
   }
 
+  async createParticipation(
+    prisonNumber: Person['prisonNumber'],
+    courseId?: Course['id'],
+    otherCourseName?: string,
+  ): Promise<CourseParticipation> {
+    return this.restClient.post({
+      data: { courseId, otherCourseName, prisonNumber },
+      path: apiPaths.participations.create({}),
+    }) as Promise<CourseParticipation>
```

Comment:
I think the `as` doesn't need the `Promise`?

Reply:
Yeah needs it otherwise you get `Conversion of type 'Promise<unknown>' to type 'CourseParticipation' may be a mistake because neither type sufficiently overlaps with the other.`. Did wonder if it might be nice to update the rest client methods to take something like `<T>` so we can specify the response. One for another PR/Spike though.

Reply:
Oh that's strange! The others seem not to wrap the `as` types in promises 🤔

Reply:
The others wrap the returned await in `()`.  Will update to be be consistent for now but be interested to investigate if that's necessary.

Reply:
Ah, just seen this after commenting in the other PR! Yeah, perhaps these can all just be promises, like in the services, but best to be consistent unless we're updating the rest